packages/js/src/settings/hooks/use-new-content-type-notification.js

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from "@wordpress/element";
+import { some } from "lodash";
+import { useDispatchSettings, useSelectSettings } from "../hooks";
+
+/**
+ * Remove new content type notification when all new content types have been reviewed.
+ * @returns {void}
+ */
+const useNewContentTypeNotification = () => {
+	const { removeNotification } = useDispatchSettings();
+	const notifications = useSelectSettings( "selectNotifications" );
+	const postTypes = useSelectSettings( "selectPostTypes" );
+	const taxonomies = useSelectSettings( "selectTaxonomies" );
+
+	useEffect( () => {
+		const newPostTypes = some( postTypes, [ "isNew", true ] );
+		const newTaxonomies = some( taxonomies, [ "isNew", true ] );
+	   if ( notifications[ "new-content-type" ] && ! newPostTypes.length && ! newTaxonomies.length ) {
+		   removeNotification( "new-content-type" );
+	   }
+	}, [ postTypes, taxonomies ] );
```

Comment:
About notifications: I'm changing the `notifications` and if I'll add it there it will create an infinite loop.
I'll add the `removeNotification` even though I don't see a reason why, but it doesn't harm.